cmd/build.go

```diff
@@ -146,12 +154,95 @@ func outputBinaryPath(projectName string, targetOS string) string {
 	return outputBinaryPath
 }
 
-func build(projectName string, targetOS string, vmArguments []string) {
-	if targetOS != runtime.GOOS {
-		fmt.Println("hover: Cross-compiling is currently not supported")
+func dockerBuild(projectName string, targetOS string, vmArguments []string) {
+	buildMode := "release"
+	if buildDebug {
+		buildMode = "debug"
+	}
+	crossCompilingDir, err := filepath.Abs(filepath.Join(buildPath, "cross-compiling", targetOS, buildMode))
+	err = os.MkdirAll(crossCompilingDir, 0755)
+	if err != nil {
+		fmt.Printf("hover: Cannot create the cross-compiling directory: %v\n", err)
+		os.Exit(1)
+	}
+	userCacheDir, err := os.UserCacheDir()
+	if err != nil {
+		fmt.Printf("hover: Cannot get the path for the system cache directory: %v\n", err)
+		os.Exit(1)
+	}
+	goPath := filepath.Join(userCacheDir, "hover-cc")
+	err = os.MkdirAll(goPath, 0755)
+	if err != nil {
+		fmt.Printf("hover: Cannot create the hover-cc GOPATH under the system cache directory: %v\n", err)
+		os.Exit(1)
+	}
+	wd, err := os.Getwd()
+	if err != nil {
+		fmt.Printf("hover: Cannot get the path for current directory %s", err)
+		os.Exit(1)
+	}
+	dockerFilePath, err := filepath.Abs(filepath.Join(crossCompilingDir, "Dockerfile"))
+	if err != nil {
+		fmt.Printf("hover: Failed to resolve absolute path for Dockerfile %s: %v\n", dockerFilePath, err)
+		os.Exit(1)
+	}
+	if _, err := os.Stat(dockerFilePath); os.IsNotExist(err) {
+		dockerFile, err := os.Create(dockerFilePath)
+		if err != nil {
+			fmt.Printf("hover: Failed to create Dockerfile %s: %v\n", dockerFilePath, err)
+			os.Exit(1)
+		}
+		dockerFileContent := []string{
+			"FROM dockercore/golang-cross",
+			"RUN apt-get install libgl1-mesa-dev xorg-dev -y",
+			"WORKDIR /app/go",
+			"CMD " + strings.Join(buildCommand(targetOS, vmArguments, "build/outputs/"+targetOS+"/"+outputBinaryName(projectName, targetOS)), " "),
```

Comment:
I don't quite understand this, why are all arguments already given in the Dockerfile? Wouldn't it be easier to add them to the `docker run` command below?